program_base.go

```diff
@@ -44,6 +44,19 @@ static int ebpf_prog_load(const char *name, __u32 prog_type, const void *insns,
 	return res;
 }
 
+static int ebpf_obj_pin(__u32 fd, const char *pathname,
```

Comment:
this reminds me `ebpf_obj_pin` from `map.go`, are they anyhow different?

Reply:
You're right

I'm going to introduce go wrapper for this function to make it reusable.